homeassistant/components/media_player/webostv.py

```diff
@@ -363,3 +381,84 @@ def media_next_track(self):
     def media_previous_track(self):
         """Send the previous track command."""
         self._client.rewind()
+
+    def play_media(self, media_type, media_id, **kwargs):
+        """Play a piece of media."""
+        self._playing = True
+        self._state = STATE_PLAYING
+        if media_type != MEDIA_TYPE_CHANNEL:
+            self._client.play()
+        else:
+            self._client.set_channel(media_id)
+
+    def update_channel_info(self):
+        channel = self._client.get_current_channel()
+
+        # no need to updated channel info while we are not watching TV
+        if channel.get('returnValue') is None:
+            self.reset_channel_info()
+            return
+
+        self._channel = channel.get('channelName')
+        self._title = self._channel
+        self._duration = None
+        channel_id = channel.get('channelId')
+        self._progress_updated_at = util.dt.utcnow()
```

Comment:
This should not be updated on each call. It should only be updated if the source is changing. Otherwise we keep getting unnecessary state updates on each update